entrypoint.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+fail () {
+    echo "$@" >&2
+    exit 1
+}
+
+shopt -s nullglob
+cvmfs_mounts=(/cvmfs/*)
+if [[ -z $cvmfs_mounts && -n $CVMFSEXEC_REPOS && -n $CVMFSEXEC_DIST ]]; then
+    echo "No cvmfs mounts; using cvmfsexec to mount $CVMFSEXEC_REPOS"
+    # We need our own copy of cvmfsexec for permissions reasons.
+    cvmfsexec_root=/tmp/cvmfsexec-$(id -u)
+    if [[ ! -d $cvmfsexec_root ]]; then
+        echo "No cvmfsexec dir found for this user at $cvmfsexec_root; creating one"
+        cp -rp /cvmfsexec $cvmfsexec_root || fail "Couldn't create $cvmfsexec_root"
+        echo "Fetching $CVMFSEXEC_DIST CVMFS config"
+        $cvmfsexec_root/makedist $CVMFSEXEC_DIST || fail "Couldn't fetch CVMFS config"
```

Comment:
I tried that, but the intent is for this image to be runnable as any UID, and I got problems when running as a different UID than the one that did the makedist.

Reply:
If you do a makedist -o cvmfsexec (in a different directory or different name so as not overwrite cvmfsexec script) during the container build, then when cvmfsexec is executed it at run time it should self-extract as the user that is executing.  Alternatively maybe you could do a chown -R on the cvmfsexec directory.

Reply:
On second thought I think using makedist -o isn't a good idea, because there ought to be a way for people to bind mount a new default.local, cache directory, and log directory.  So it's better to have it be unpacked.  However I don't understand why the cp -r to /tmp didn't make everything writable.

Reply:
makedist creates dist/var/lib/cvmfs with 0700 permissions so cp -r can't read it.

Reply:
That comes from the cvmfs rpm so I guess you'll have to either chmod it, rmdir it and recreate it on the destination, or use a copy mechanism where you can exclude it.

Copying all the files to a new location complicates allowing people to bind mount in cache and log directories too.  I think it would probably be best to skip both dist/var/lib/cvmfs and dist/var/log and manipulate them manually.  If somebody has bind mounted either one at a documented location, create a symlink pointing back to them, otherwise I guess leave them in /tmp.  By looking at `mount` inside of a docker container it looks like the whole root filesystem is an overlay with the layers in /var/lib/docker/overlay2 so I guess that's OK as a default, but /var has to be big enough.  I think it would make sense to support bind-mounting at the paths /var/lib/cvmfs and /var/log/cvmfs, and also /etc/cvmfs/default.local.